ios/HybridImageFactory.swift

```diff
@@ -21,14 +22,102 @@ class HybridImageFactory: HybridImageFactorySpec {
   }
   
+  /**
+   * Load Image from URL
+   */
+  func loadFromAssetAsync(assetId: String, options: AssetImageLoadOptions?) throws -> Promise<any HybridImageSpec> {
+    let assets = PHAsset.fetchAssets(
+      withLocalIdentifiers: [assetId],
+      options: nil
+    )
+    guard let asset = assets.firstObject else {
+      throw NSError(
+        domain: "TurboImageView",
+        code: 404,
+        userInfo: [NSLocalizedDescriptionKey: "Asset not found"]
+      )
+    }
+    
+    return Promise.async {
+      return try await withCheckedThrowingContinuation { continuation in
+        let requestOptions = PHImageRequestOptions()
+        requestOptions.version = .current
+        requestOptions.deliveryMode = .highQualityFormat
+        requestOptions.isNetworkAccessAllowed = true
+        
+        
+        if let size = options?.size {
+          let contentMode: PHImageContentMode
+          if let aspectFit = options?.aspectFit {
+            switch aspectFit {
+            case .fit:
+              contentMode = .aspectFit
+            case .fill:
+              contentMode = .aspectFill
+            }
+          } else {
+            contentMode = .default
+          }
+          
+          PHImageManager.default().requestImage(
+            for: asset,
+            targetSize: CGSize(width: size.width, height: size.height),
+            contentMode: contentMode,
+            options: requestOptions
+          ) { (image, info) in
+            if let error = info?[PHImageErrorKey] as? Error {
+              continuation.resume(throwing: error)
+            } else if let image = image {
+              continuation
+                .resume(returning: HybridImage(uiImage: image))
+            } else {
+              continuation.resume(
+                throwing: RuntimeError.error(withMessage: "Failed to load or convert image")
+              )
+            }
+          }
+        } else {
+          PHImageManager.default().requestImageDataAndOrientation(
+            for: asset,
+            options: requestOptions
+          ) { (imageData, dataUTI, orientation, info) in
+            
+            if let error = info?[PHImageErrorKey] as? Error {
+              continuation.resume(throwing: error)
+            } else if let imageData = imageData,
+                      let cgImage = UIImage(data:imageData)?.cgImage                       {
+              let uiImageOrientation: UIImage.Orientation
+              switch orientation {
+              case .up: uiImageOrientation = .up
+              case .upMirrored: uiImageOrientation = .upMirrored
+              case .down: uiImageOrientation = .down
+              case .downMirrored: uiImageOrientation = .downMirrored
+              case .left: uiImageOrientation = .left
+              case .leftMirrored: uiImageOrientation = .leftMirrored
+              case .right: uiImageOrientation = .right
+              case .rightMirrored: uiImageOrientation = .rightMirrored
+              }
```

Comment:
Also create an extension for this. Why pollute the `HybridImageFactory` when we can nicely move this into extension methods that have clear naming as to what they do internally?